modular_zubbers/modules/borgs/code/robot_upgrade.dm

```diff
@@ -54,3 +54,39 @@
 	model_flags += BORG_MODEL_RESEARCH
 	items_to_remove = list(/obj/item/surgical_drapes)
 
+
+/// Quadruped special module
+/obj/item/borg/upgrade/squadrupedmodule
+	name = "small quadruped borg affection module"
+	desc = "A module that upgrades the ability of small quadruped borgs to display affection."
+	icon_state = "cyborg_upgrade3"
+
+// To prevent nonsmalls having this module this is method is depreciated
+/obj/item/borg/upgrade/squadrupedmodule/action(mob/living/silicon/robot/borg)
+	. = ..()
+	if(!.)
+		return
+	if(borg.hasAffection)
+		to_chat(usr, span_warning("This unit already has a affection module installed!"))
+		return FALSE
+	if(!(TRAIT_R_SQUADRUPED in borg.model.model_features))
+		to_chat(usr, span_warning("This unit's chassis does not support this module."))
+		return FALSE
+
+	var/obj/item/quadborg_tongue/quadtongue = new /obj/item/quadborg_tongue(borg.model)
+	borg.model.basic_modules += quadtongue
+	borg.model.add_module(quadtongue, FALSE, TRUE)
+	var/obj/item/quadborg_nose/quadnose = new /obj/item/quadborg_nose(borg.model)
+	borg.model.basic_modules += quadnose
+	borg.model.add_module(quadnose, FALSE, TRUE)
+	borg.hasAffection = TRUE
+
+/obj/item/borg/upgrade/squadrupedmodule/deactivate(mob/living/silicon/robot/borg, user = usr)
+	. = ..()
+	if(.)
+		return
+	borg.hasAffection = FALSE
+	for(var/obj/item/quadborg_tongue/quadtongue in borg.model.modules)
+		borg.model.remove_module(quadtongue, TRUE)
+	for(var/obj/item/quadborg_nose/quadnose in borg.model.modules)
+		borg.model.remove_module(quadnose, TRUE)
```

Comment:
There is a reason I did this and that is the new method is not compatible with checking for traits this method is depreciated and should never be used else where.